test/e2e/common/secrets.go

```diff
@@ -313,3 +317,80 @@ func doSecretE2EWithMapping(f *framework.Framework, mode *int32) {
 
 	f.TestContainerOutput("consume secrets", pod, 0, expectedOutput)
 }
+
+//Existence of a second secret with same name and different namespace shouldn't affect the mount
+func doSecretE2EForMountNameSpaceIssue(f *framework.Framework, defaultMode *int32) {
```

Comment:
But that is needed only because it's run with the flag to print the file mode. If we are not testing permissions here, why have it? I mean, it's always used as nil this param. And except we want to test that the secret in the right ns is mounted AND the permission is respected in the same test, I don't see the point. Am I missing something?

If it's because code is shared and is more easy to test both, it seems fine to me. But if it's just "duplicated" code, I think it's okay to just test the permission in another test. What do you think?